### ISSUE TITLE
Beam deconvolution is attempted on masked out channels

### ISSUE THREAD

**Issue by keflavich** [radio-astro-tools/spectral-cube#436]
In convolve_to:

```
mcube = cube.mask_out_bad_beams(0.1)

In [18]: mcube[:5].beams
Out[18]:
[Beam: BMAJ=3.158775568008423 arcsec BMIN=2.0589499473571777 arcsec BPA=68.22599792480469 deg,
 Beam: BMAJ=3.159079074859619 arcsec BMIN=2.058413505554199 arcsec BPA=68.24202728271484 deg,
 Beam: BMAJ=0.5349162817001343 arcsec BMIN=0.44836196303367615 arcsec BPA=58.74653244018555 deg,
 Beam: BMAJ=0.5350565314292908 arcsec BMIN=0.4485272765159607 arcsec BPA=58.651527404785156 deg,
 Beam: BMAJ=0.5351839661598206 arcsec BMIN=0.4486926198005676 arcsec BPA=58.64647674560547 deg]

In [15]: mcube[:5].convolve_to(radio_beam.Beam(1*u.arcsec))
Traceback (most recent call last):
  File "<ipython-input-15-d180867ca7ab>", line 1, in <module>
    mcube[:5].convolve_to(radio_beam.Beam(1*u.arcsec))
  File "/home/ginsburg/anaconda3/lib/python3.5/site-packages/spectral_cube/spectral_cube.py", line 3216, in convolve_to
    cb = beam.deconvolve(bm)
  File "/home/ginsburg/repos/radio_beam/build/lib.linux-x86_64-3.5/radio_beam/beam.py", line 379, in deconvolve
    raise ValueError("Beam could not be deconvolved")
ValueError: Beam could not be deconvolved
```

**Comment by keflavich** [comment]
Tricky thing: `beams` is a list, and it is used by both `SpectralCube` and `LowerDimensionalObject`s.  We need a mask that gets recognized by both.

**Comment by keflavich** [comment]
https://github.com/radio-astro-tools/radio-beam/pull/51 is an incomplete solution that we'll hopefully complete soon.

**Comment by e-koch** [comment]
Fixed in #437
